app/assets/javascripts/questions.js

```diff
@@ -9,6 +9,60 @@ $(document).ready(function(q){
       $('.my-question').addClass('hidden');
       $('.other-question').removeClass('hidden');
     });
+  
+// Defines function that disables/enables submit button depending on if there is text in the answer box
+function enableSubmitAnswer() {
+  if ($('#answer_text').val()) {
+    $('#submit_answer').attr('disabled', false);
+  } else {
+    $('#submit_answer').attr('disabled', true); 
+  }
+}
+
+// Defines function that gives the user feedback once they state whether or not they have gotten the question right. 
+// This function also stores the user response in the development_analytics.log using an ajax call. 
+ function provide_feedback(is_correct, notice_text, qid) {
+   $.ajax({
+    url: window.location.pathname +  "/" + q_id + "/feedback",
+     type: "POST",
+        data: { correct: is_correct }
+    })
+    .done(function( data ) {
+      if ( console && console.log ) {
+        console.log( "Sample of data:", data.slice( 0, 100 ) );
+      }
+      $('#alert_text').text(notice_text);
+      $('.alert').slideDown();
+    });
+  };
+
+// Creates the javscript for the modal that allows it to be populated with the data recieved upon clicking the modal link
+$('#question_display_Modal').on('show.bs.modal', function (event) {
+  
+  // Creates the initial view of the modal. The submit answer is shown as well as an empty text box. 
+  // The answer is initally hidden and so is the response + alert forms. 
+  $("#submit_answer").show();
+  $('#answer_text').val(""); 
+  enableSubmitAnswer(); 
+  $("#answers").hide();
+  $("#alert_text_container").hide();
+
+  var question_link = $(event.relatedTarget) // Button that triggered the modal. question_link is populated with the data from the click 
```

Comment:
This might not be a `Button` (in fact, it is a link/`a` tag at the moment). So perhaps rephrasing this as `element that triggered the modal` would be less confusing.